e2e/cypress/integration/references.test.js

```diff
@@ -0,0 +1,233 @@
+        .children()
+        .should('have.length', 4);
+    });
+
+    it('displays empty search results if item doesnt have references in inspire', () => {
+      cy.on('uncaught:exception', () => {
+        return false;
+      });
+
+      cy.registerRoute();
+      cy.visit('/literature/1787272');
+      cy.waitForRoute();
+
+      cy.get('[data-test-id="reference-search-button"]').click();
+
+      cy.get('.ant-empty').should('be.visible');
+    });
+  });
+
+  describe('Reference container', () => {
+    onlyOn('headless', () => {
+      it('number of references per page', () => {
+        cy.on('uncaught:exception', (err, runnable) => {
+          return false;
+        });
+        cy.registerRoute();
+        cy.visit('/literature/1374620');
+        cy.waitForRoute();
+
+        cy.get('[data-test-id="pagination-list"]').as('paginationList');
+        cy.get('@paginationList')
+          .find('.ant-list-items')
+          .children()
+          .as('referenceListItems');
+        cy.get('@paginationList')
+          .find('span[class="ant-select-selection-item"]')
+          .as('selectionItem');
+
+        cy.get('@referenceListItems').should('have.length', 25);
+        cy.get('@selectionItem').should('be.visible');
+
+        cy.get('@paginationList')
+          .find('.ant-select-selection-search-input')
+          .click({ force: true });
+        cy.get('@paginationList')
+          .find('div')
+          .contains('50 / page')
+          .click({ force: true });
+        cy.waitForRoute();
+
+        cy.get('@referenceListItems').should('have.length', 50);
+        cy.get('@selectionItem').should('be.visible');
+        cy.get('a[href="/literature"]').click();
+        cy.waitForRoute();
+
+        cy.get('a[href="/literature/1598135"]').click();
+        cy.waitForRoute();
+        cy.get('@referenceListItems').should('have.length', 50);
+      });
+    });
+
+    describe('Self curation', () => {
+      it('edit button is disabled when user is not logged in', () => {
+        cy.visit('/literature/1688995');
+        cy.waitForLoading();
+
+        cy.get('[data-test-id="edit-reference"]').first().find('.ant-btn').should('be.disabled');
+      });
+
+      it('curates reference successfully', () => {
+        cy.login('cataloger');
+
+        cy.registerRoute();
+        cy.visit('/literature/1688995');
+        cy.waitForLoading();
+
+        cy.get('[data-test-id="edit-reference"]').first().click();
+        cy.get('[data-test-id="reference-embedded-search')
+          .find('.ant-input-search-button')
+          .click();
+        cy.waitForLoading();
+
+        cy.get('[data-test-id="reference-drawer-item-1787272"]').click();
+        cy.get('[data-test-id="curate-button"').click();
+
+        cy.waitForLoading();
+        cy.wait(5000);
+
+        cy.get('.ant-notification-notice-message').should(
+          'have.text',
+          'Reference Successfully Modified!'
+        );
+      });
```

Comment:
is it possible to check if the reference text changed?

Reply:
added